botcore/utils/members.py

```diff
@@ -0,0 +1,56 @@
+"""Useful helper functions for interactin with :obj:`discord.Member` objects."""
+
+import typing
+
+import discord
+
+from botcore.utils import loggers
+
+log = loggers.get_logger(__name__)
+
+
+async def get_or_fetch_member(guild: discord.Guild, member_id: int) -> typing.Optional[discord.Member]:
+    """
+    Attempt to get a member from cache; on failure fetch from the API.
+
+    Returns:
+        The :obj:`discord.Member` or :obj:`None` to indicate the member could not be found.
+    """
+    if member := guild.get_member(member_id):
+        log.trace(f"{member} retrieved from cache.")
+    else:
+        try:
+            member = await guild.fetch_member(member_id)
+        except discord.errors.NotFound:
+            log.trace(f"Failed to fetch {member_id} from API.")
+            return None
+        log.trace(f"{member} fetched from API.")
+    return member
+
+
+async def handle_role_change(
+    member: discord.Member,
+    coro: typing.Callable[..., typing.Coroutine],
+    role: discord.Role
+) -> None:
+    """
+    Await the given ``coro`` with ``member`` as the sole argument.
```

Comment:
If we're going to use this style, then it needs to be consistently applied. This uses double backticks while other docstrings only have single. Not sure which is better.

Reply:
https://github.com/python-discord/bot-core/pull/29/commits/d7841816581737b4a7cece5bba98328a44630d4b